core-container/entry.sh

```diff
@@ -127,7 +140,7 @@ function wait-kubestellar-ready() {
 
 
 function guess_kcp_dns() {
-    KUBECONFIG=
+    KUBECONFIG=$SPACE_MANAGER_KUBECONFIG
```

Comment:
You assume here and in many other places that the SPACE_MANAGER_KUBECONFIG == access to the hosting cluster. I think this is a private case that should be configurable and not hard-coded. It can also be the same cluster with different user and permissions 
E.e., the SM config allows to access space related resources and not "routes"